dataland_analytics/analysis_scripts/scripts_with_api_admin/requests.py

```diff
@@ -0,0 +1,166 @@
+
+    return [req for req in requests_by_criteria]
+
+
+def get_pending_datasets_for_requests(requests: list[ExtendedStoredDataRequest]) -> list[ReviewQueueResponse]:
+    """
+    This function filters out from the list of pending datasets all those which match a request of the input
+    :param requests: requests to find the pending datasets for
+    :return: list of matching datasets
+    """
+    pending_datasets: list[ReviewQueueResponse] = get_pending_datasets()
+
+    matching_datasets = []
+    for dataset_metadata in pending_datasets:
+        for req in requests:
+            if is_match(dataset_metadata, req):
+                matching_datasets.append(dataset_metadata)
+
+    return matching_datasets
+
+
+def filter_requests_updated_later_than_threshold(requests: list[ExtendedStoredDataRequest], timestamp: datetime)\
+        -> list[ExtendedStoredDataRequest]:
+    """
+    Filter list of requests by timestamp
+    :param requests: list of requests
+    :param timestamp: datetime threshold
+    :return: list of matching ExtendedStoredDataRequests
+    """
+    return list(filter(lambda req: timestamp_in_ms_to_datetime(req.last_modified_date) > timestamp, requests))
+
+
+def is_match(data: ReviewQueueResponse, req: ExtendedStoredDataRequest) -> bool:
+    """
+    Returns true if dataset matches the request
+    :param data: dataset
+    :param req: request
+    :return: bool
+    """
+    return (data.framework == req.data_type and data.reporting_period == req.reporting_period
+            and data.company_id == req.dataland_company_id)
+
+
+def format_request_for_export(*requests: ExtendedStoredDataRequest) -> list[dict]:
+    keys_map = {
+        'dataRequestId': 'Request ID',
+        'userEmailAddress': 'Requester',
+        'creationTimestamp': 'Date of Request',
+        'dataType': 'Framework',
+        'reportingPeriod': 'Reporting Period',
+        'datalandCompanyId': 'Company ID',
+        'companyName': 'Company',
+        'lastModifiedDate': 'Last Update of Request',
+        'requestStatus': 'Status',
+    }
+    values_map = {
+        'dataType': get_readable_datatype_string_from_input,
+        'creationTimestamp': timestamp_in_ms_to_datetime,
+        'lastModifiedDate': timestamp_in_ms_to_datetime,
+        'requestStatus': get_enum_value,
+    }
+
+    return [{keys_map[key]: values_map[key](value) if key in values_map else value
+            for key, value in req.to_dict().items() if key in keys_map} for req in requests]
+
+
+def export_requests_to_excel(requests: list[dict], path: str, exclude_columns=None):
+    if len(requests) > 0:
+        requests_dataframe = pd.DataFrame.from_records(requests, exclude=exclude_columns)
+        with pd.ExcelWriter(path=path, engine="openpyxl",
+                            mode='w', datetime_format='YYYY-MM-DD HH:MM:SS') as excel_writer:
+            requests_dataframe.to_excel(excel_writer, sheet_name="Overview")
+        adjust_excel_column_widths(file_name)
```

Comment:
scope of filename variable not defined within this function, could result in a NameError. Maybe replace with

`        adjust_excel_column_widths(path)
`

Reply:
done